packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -63,6 +63,8 @@ export class AwsCliCompatible {
 
     if (options.containerCreds ?? hasEcsCredentials()) {
       sources.push(() => new AWS.ECSCredentials());
+    } else if (options.containerCreds ?? hasWebIdentityCredentials() ) {
```

Comment:
I'm not quite parsing what this conditional intends to convey (the relationship between `containerCreds` and the auto-detect).

Can you describe in words when this branch should be taken?

As it is, I think `if (hasWebIdentityCredentials() && options.containerCreds !== false)` is equivalent to what you wrote here and more understandable (imo), but I'm not sure that's what you intended.